File: app/src/main/java/es/etologic/mahjongscoring2/data/repositories/BaseRepository.java
package es.etologic.mahjongscoring2.data.repositories;

import android.content.Context;

import javax.inject.Singleton;

import es.etologic.mahjongscoring2.data.local_data_source.local.AppDatabase;

@Singleton
public abstract class BaseRepository {

    protected final AppDatabase database;

    BaseRepository(Context context) {
        database = AppDatabase.getInstance(context.getApplicationContext());
    }
}

File: app/src/main/java/es/etologic/mahjongscoring2/data/repositories/BaseRepository.kt
package es.etologic.mahjongscoring2.data.repositories

import android.content.Context
import es.etologic.mahjongscoring2.data.local_data_source.local.AppDatabase
import javax.inject.Singleton

@Singleton
abstract class BaseRepository internal constructor(context: Context) {
    
    protected val database: AppDatabase? = AppDatabase.getInstance(context.applicationContext)
}
